### PR TITLE
Frontend: Muted conversations

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers CC @mlsteele

Here's our conversation muting support.  New behavior:

* If you run `keybase chat mute someone`, you'll see a `shh` next to their entry in the sidebar.  The conversation snippet will go away from the sidebar.  You won't get desktop notifications for new messages from them anymore.
* If you run `keybase chat mute -u someone`, the reverse will be true.
* If you open the sidebar (information icon) for your conversation, there's an "Mute notifications" checkbox that has the same effect as switching between `mute` and `mute -u`.

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
The commits 27afe20bcd07...8df80f165d65 introduce visual changes on linux. <details><summary>:mag_right: 3 changed</summary>

 * Changed: **Menubar-Private_Async_Writing** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1486075857681-27afe20bcd07-8df80f165d65-linux/Menubar-Private_Async_Writing-27afe20bcd07.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1486075857681-27afe20bcd07-8df80f165d65-linux/Menubar-Private_Async_Writing-8df80f165d65.png)  
   Menubar-Private_Async_Writing comparison
 * Changed: **Chat_Side_Panel-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1486075857681-27afe20bcd07-8df80f165d65-linux/Chat_Side_Panel-Normal-27afe20bcd07.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1486075857681-27afe20bcd07-8df80f165d65-linux/Chat_Side_Panel-Normal-8df80f165d65.png)  
   Chat_Side_Panel-Normal comparison
 * Changed: **Chat_Side_Panel-Empty** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1486075857681-27afe20bcd07-8df80f165d65-linux/Chat_Side_Panel-Empty-27afe20bcd07.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1486075857681-27afe20bcd07-8df80f165d65-linux/Chat_Side_Panel-Empty-8df80f165d65.png)  
   Chat_Side_Panel-Empty comparison
</summary>

File: shared/actions/chat.js [hunk 1]
```diff
@@ -235,6 +239,10 @@ function editMessage (message: Message, text: HiddenString): EditMessage {
   return {type: 'chat:editMessage', payload: {message, text}}
 }
 
+function muteConversation (conversationIDKey: ConversationIDKey, muted: boolean) {
```
Comment (MarcoPolo) [inline shared/actions/chat.js]: This should return `Constants.MuteConversation`

File: shared/actions/chat.js [hunk 2]
```diff
@@ -1233,6 +1258,16 @@ function * _selectConversation (action: SelectConversation): SagaGenerator<any,
   }
 }
 
+function * _muteConversation (action: MuteConversation): SagaGenerator<any, any> {
+  const {conversationIDKey, muted} = action.payload
+  const conversationID = keyToConversationID(conversationIDKey)
+  const status = muted ? CommonConversationStatus.muted : CommonConversationStatus.unfiled
+  const identifyBehavior: TLFIdentifyBehavior = 2 // CHAT_GUI
```
Comment (MarcoPolo) [inline shared/actions/chat.js]: Why not use `flowTypes.TlfKeysTLFIdentifyBehavior.chatGui` instead of `2`

**Comment by cjb** [comment]
@MarcoPolo Thanks, did both of those!

**Comment by keybase-ci-visdiff** [comment]
The commits 27afe20bcd07...36ed65b9eecc introduce visual changes on linux. <details><summary>:mag_right: 3 changed</summary>

 * Changed: **unlock_folders-Success** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1486138398093-27afe20bcd07-36ed65b9eecc-linux/unlock_folders-Success-27afe20bcd07.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1486138398093-27afe20bcd07-36ed65b9eecc-linux/unlock_folders-Success-36ed65b9eecc.png)  
   unlock_folders-Success comparison
 * Changed: **Chat_Side_Panel-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1486138398093-27afe20bcd07-36ed65b9eecc-linux/Chat_Side_Panel-Normal-27afe20bcd07.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1486138398093-27afe20bcd07-36ed65b9eecc-linux/Chat_Side_Panel-Normal-36ed65b9eecc.png)  
   Chat_Side_Panel-Normal comparison
 * Changed: **Chat_Side_Panel-Empty** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1486138398093-27afe20bcd07-36ed65b9eecc-linux/Chat_Side_Panel-Empty-27afe20bcd07.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1486138398093-27afe20bcd07-36ed65b9eecc-linux/Chat_Side_Panel-Empty-36ed65b9eecc.png)  
   Chat_Side_Panel-Empty comparison
</summary>

**Pull request merged**